File: app/src/main/java/org/mifos/mobile/utils/FaqDiffUtil.java
package org.mifos.mobile.utils;

import org.mifos.mobile.models.FAQ;

import java.util.List;

import androidx.recyclerview.widget.DiffUtil;

/**
 * Created by dilpreet on 12/8/17.
 */

public class FaqDiffUtil extends DiffUtil.Callback {

    private List<FAQ> oldFaq;
    private List<FAQ> newFaq;

    public FaqDiffUtil(List<FAQ> oldFaq, List<FAQ> newFaq) {
        this.oldFaq = oldFaq;
        this.newFaq = newFaq;
    }

    @Override
    public int getOldListSize() {
        return oldFaq.size();
    }

    @Override
    public int getNewListSize() {
        return newFaq.size();
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        return oldFaq.get(oldItemPosition).getQuestion().
                compareTo(newFaq.get(newItemPosition).getAnswer()) == 0;
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        return oldFaq.get(oldItemPosition).equals(newFaq.get(newItemPosition));
    }
}

File: app/src/main/java/org/mifos/mobile/utils/FaqDiffUtil.kt
package org.mifos.mobile.utils

import androidx.recyclerview.widget.DiffUtil
import org.mifos.mobile.models.FAQ

/**
 * Created by dilpreet on 12/8/17.
 */
class FaqDiffUtil(private val oldFaq: ArrayList<FAQ?>??, private val newFaq: ArrayList<FAQ?>?) : DiffUtil.Callback() {
    override fun getOldListSize(): Int {
        return if (oldFaq?.size != null) oldFaq.size
        else 0
    }

    override fun getNewListSize(): Int {
        return if (newFaq?.size != null) newFaq.size
        else 0
    }

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return newFaq?.get(newItemPosition)?.answer?.let { oldFaq?.get(oldItemPosition)?.question?.compareTo(it) } == 0
    }

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return oldFaq?.get(oldItemPosition) == newFaq?.get(newItemPosition)
    }
}